spynnaker/pyNN/models/neuron/synapse_types/synapse_type_exponential.py

```diff
@@ -39,16 +40,18 @@ def data_type(self):
 def get_exponential_decay_and_init(tau, machine_time_step):
     ulfract = pow(2, 32)
     ts = float(machine_time_step) / 1000.0
-    return (
-
-        # decay = e^(-ts / tau) as an unsigned long fract
-        tau.apply_operation(
-            lambda x: int(numpy.exp(-ts / x) * ulfract)),
-
-        # init = (tau / ts) * (1 - e^(-ts / tau)) as an unsigned long fract
-        tau.apply_operation(
-            lambda x: int((x / ts) * (1.0 - numpy.exp(-ts / x)) * ulfract))
-    )
+    # decay = e^(-ts / tau) as an unsigned long fract
+    decay = lambda x: int(numpy.exp(-ts / x) * ulfract)  # noqa E731
+    # init = (tau / ts) * (1 - e^(-ts / tau)) as unsigned long fract
+    init = lambda x: int((x / ts) * (1.0 - numpy.exp(-ts / x)) * ulfract)  # noqa E731,
```

Comment:
See above re: removing noqa and replacing with a function.

Reply:
Having discussed this with Christian, I withdraw the objection, as the lambda function is dependent on other functions of values, which would make the code less readable.